lib/core/src/display/shape/text/text_field.rs

```diff
@@ -27,6 +27,25 @@ use crate::display::world::World;
 use nalgebra::Vector2;
 use nalgebra::Vector3;
 use nalgebra::Vector4;
+use std::ops::Range;
+
+
+
+// ===============================
+// === TextChangedNotification ===
+// ===============================
+
+/// A notification about text change.
+///
+/// In essence, it's `TextChange` with some additional useful information.
+#[derive(Clone,Debug,Shrinkwrap)]
+pub struct TextChangedNotification {
+    /// A change which has occurred.
+    #[shrinkwrap(main_field)]
+    pub change : TextChange,
+    /// The replaced range as char positions from document begin, instead of row:column pairs.
+    pub replaced_chars : Range<usize>,
```

Comment:
Is it possible to use types from `data::text` here?
BTW if we can have Range<data::text::Index> then perhaps it should replace `data::text::Span`?

Reply:
Had a call: Deferred until future discussion about how data::text is used is concluded.